cmake/MbedTargets.cmake

```diff
@@ -0,0 +1,1299 @@
+set(MBED_AVAILABLE_TARGETS "DISCO_L053C8" "NRF51_MICROBIT_BOOT" "ARCH_BLE" "RBLAB_NRF51822" "LPC1549" "RZ_A1H" "OC_MBUINO" "ARM_MPS2_M0" "ARM_MPS2_M1" "ARM_MPS2_M3" "ARM_MPS2_M4" "ARM_MPS2_M7" "STM32F407" "EFM32PG_STK3401" "ARCH_MAX" "NUCLEO_F302R8" "MAXWSNENV" "KL05Z" "NRF51_MICROBIT" "LPC4337" "DISCO_F746NG" "NUCLEO_F334R8" "NUCLEO_F411RE" "WALLBOT_BLE" "ARCH_PRO" "NUCLEO_L053R8" "DISCO_F303VC" "DISCO_F401VC" "K20D50M" "DISCO_F407VG" "LPC2460" "KL43Z" "K22F" "TY51822R3_BOOT" "ARCH_GPRS" "NRF51822_OTA" "RBLAB_NRF51822_OTA" "NUCLEO_F401RE" "NUCLEO_F303RE" "MOTE_L152RC" "LPC11U24" "DISCO_L476VG" "LPC11U34_421" "STM32F3XX" "UBLOX_C027" "NRF51822_Y5_MBUG" "NRF51822" "ARCH_BLE_BOOT" "UBLOX_C029" "TY51822R3_OTA" "SEEED_TINY_BLE" "NUCLEO_F091RC" "LPC11U24_301" "DISCO_F334C8" "RBLAB_NRF51822_BOOT" "TEENSY3_1" "LPC4088_DM" "TY51822R3" "MAX32600MBED" "RBLAB_BLENANO" "NRF51822_BOOT" "ELMO_F411RE" "WALLBOT_BLE_BOOT" "B96B_F446VE" "LPC4330_M0" "RBLAB_BLENANO_BOOT" "LPC1347" "LPC4330_M4" "DELTA_DFCM_NNN40_OTA" "LPC11C24" "DELTA_DFCM_NNN40" "SEEED_TINY_BLE_OTA" "LPC824" "MTS_MDOT_F411RE" "SAMD21J18A" "KL46Z" "SAML21J18A" "ARCH_LINK_OTA" "NRF51_DK_BOOT" "NRF51_DONGLE" "EFM32HG_STK3400" "LPC11U68" "LPC11U35_401" "DELTA_DFCM_NNN40_BOOT" "LPCCAPPUCCINO" "XADOW_M0" "LPC11U37H_401" "NRF51_MICROBIT_B_OTA" "XBED_LPC1768" "NUCLEO_L152RE" "DISCO_F429ZI" "KL26Z" "EFM32LG_STK3600" "LPC810" "LPC812" "NRF51_MICROBIT_B_BOOT" "NUCLEO_F070RB" "LPC1114" "NUCLEO_F072RB" "SEEED_TINY_BLE_BOOT" "MICRONFCBOARD" "MTS_DRAGONFLY_F411RE" "WIZWIKI_W7500ECO" "EFM32WG_STK3800" "ARM_MPS2_M0P" "DISCO_F051R8" "NUCLEO_F446RE" "ARM_MPS2_BEID" "NUCLEO_F410RB" "NUCLEO_F042K6" "NUCLEO_F103RB" "EFM32ZG_STK3200" "EFM32GG_STK3700" "NUCLEO_L476RG" "ARM_MPS2" "ARCH_LINK" "NRF51_MICROBIT_B" "NUCLEO_F030R8" "NRF51_DK_OTA" "RBLAB_BLENANO_OTA" "DISCO_F100RB" "NUCLEO_F303K8" "LPC11U35_501_IBDAP" "WIZWIKI_W7500P" "K64F" "LPC11U35_501" "NZ32SC151" "LPC2368" "ARCH_LINK_BOOT" "WALLBOT_BLE_OTA" "ARCH_BLE_OTA" "NUCLEO_L073RZ" "MTS_GAMBIT" "HRM1017" "LPC1768" "LPC4088" "WIZWIKI_W7500" "SAMD21G18A" "NRF51_DONGLE_OTA" "SAMR21G18A" "NRF51_DONGLE_BOOT" "NRF51_MICROBIT_OTA" "LPC11U35_Y5_MBUG" "LPC11U37_501" "DISCO_F469NI" "NRF51_DK" "HRM1017_BOOT" "HRM1017_OTA" "KL25Z" "NUCLEO_F031K6" "SSCI824" "MTS_MDOT_F405RG")
```

Comment:
It's generated from a python script I wrote. It looked like it was going to be tricky figuring out how it could be generated before cmake runs, so I just threw a copy in there for now.